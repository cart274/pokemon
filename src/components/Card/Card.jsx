import React from 'react';
import styles from './Card.module.css';
import notAvailableImage from '../../assets/images/imageNotAvailable.jpg'
import uuid from 'react-uuid'

const Card = ({name, id, t, onSelect, detail = {}, selectedClass, notSelectedClass, onCloseDetail}) => (
    <article className={`${styles.card} ${selectedClass && styles.cardSelected} ${notSelectedClass && styles.cardNotSelected}`}
    onClick={()=>{ !selectedClass && onSelect()}} >
        <div className={styles.description}>
            <img src={`https://pokeres.bastionbot.org/images/pokemon/${id}.png`} alt={name} 
             onError={(e)=>{e.target.onerror = null; e.target.src=notAvailableImage}} />
            <h2>{name}</h2>
        </div>
        {detail.abilities && detail.abilities.length > 0 && <div className={styles.details}>
            <h3>{`${t('Altura')}:`} <span>{`${detail.height / 10}m`}</span></h3>
            <h3>{`${t('Peso')}:`} <span>{`${detail.weight / 10}kg`}</span></h3>
            { detail.abilities.map(({ability, is_hidden}, key)=> 
                is_hidden? <h3 key={uuid()}>{`${t('Habilidad normal')}:`} <span>{ability.name}</span></h3> :
                <h3 key={uuid()}>{`${t('Habilidad oculta')}:`} <span>{ability.name}</span></h3>
            )}
            <h4 onClick={()=>{onCloseDetail()}}>{t('Cerrar')}</h4>
        </div>}
    </article>
)
export default Card