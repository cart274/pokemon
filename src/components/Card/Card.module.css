@value colors: "../../assets/styles/variables.css";
@value yellow, blue, white from colors;

.card {
    width: 14%;
    box-sizing: border-box;
    margin: 20px;
    display: flex;
    cursor: pointer;
}

.description{
  border: 3px solid yellow;
  padding: 20px 20px 10px 20px;
}

.card img{
  width: 100%;
  height: auto;
}
.card h2{
  text-align: center;
  text-transform: capitalize;
  line-height: 20px;
  font-size: 18px;
  font-weight: bold;
  color: blue;
  padding-top: 10px;
}

.details{
  padding: 20px;
  font-size: 14px;
  border: 3px solid yellow;
  border-left: none;
  width: 50%;
}
.details h3{
  padding: 5px 0;
}
.details h3 span{
  font-weight: bold;
}
.cardSelected{
  width: 40%;
}

.cardSelected .description{

}

.cardSelected .details{
}