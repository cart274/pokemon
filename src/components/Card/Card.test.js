import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('<Card />', () => {

    it('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshots', () => {
        const wrapper = shallow(<Card/>);
        expect(wrapper).toMatchSnapshot();
    });
})
