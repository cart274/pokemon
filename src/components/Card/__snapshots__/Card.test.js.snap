// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="description"
        >
          <img
            onError={[Function]}
            src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
          />
          <h2 />
        </div>,
        undefined,
      ],
      "className": "card undefined undefined",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              onError={[Function]}
              src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
            />,
            <h2 />,
          ],
          "className": "description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "onError": [Function],
              "src": "https://pokeres.bastionbot.org/images/pokemon/undefined.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="description"
          >
            <img
              onError={[Function]}
              src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
            />
            <h2 />
          </div>,
          undefined,
        ],
        "className": "card undefined undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                onError={[Function]}
                src="https://pokeres.bastionbot.org/images/pokemon/undefined.png"
              />,
              <h2 />,
            ],
            "className": "description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "onError": [Function],
                "src": "https://pokeres.bastionbot.org/images/pokemon/undefined.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
